api/millenium_patron.py

```diff
@@ -78,15 +86,26 @@ def remote_authenticate(self, username, password):
         valid, a PatronData that serves only to indicate which
         authorization identifier the patron prefers.
         """
-        path = "%(barcode)s/%(pin)s/pintest" % dict(
-            barcode=username, pin=password
-        )
-        url = self.root + path
-        response = self.request(url)
-        data = dict(self._extract_text_nodes(response.content))
-        if data.get('RETCOD') == '0':
-            return PatronData(authorization_identifier=username, complete=False)
-        return False
+        if self.auth_mode == "pin":
+            path = "%(barcode)s/%(pin)s/pintest" % dict(
+                barcode=username, pin=password
+            )
+            url = self.root + path
+            response = self.request(url)
+            data = dict(self._extract_text_nodes(response.content))
+            if data.get('RETCOD') == '0':
+                return PatronData(authorization_identifier=username, complete=False)
+            return False
+        elif self.auth_mode == "last_name":
+            path = "%(barcode)s/dump" % dict(barcode=username)
+            url = self.root + path
+            response = self.request(url)
+            data = dict(self._extract_text_nodes(response.content))
+            dump_name = data.get(self.PERSONAL_NAME_FIELD)
+            dump_name = dump_name.split(',')[0]
+            if dump_name == password:
```

Comment:
This needs to be a case-insensitive comparison, because the names are stored in uppercase and most people will enter their name in lowercase.